### PR TITLE
Match the standard NumPy convention in JAX (onp -> np)

### PR DESCRIPTION
**Pull request by 8bitmp3**
Switching away from using `onp` for ordinary NumPy towards `np`, as discussed with @avital in https://github.com/google/flax/discussions/730.

This is in line with JAX https://github.com/google/jax/pull/3760 (by @jakevdp) and https://github.com/google/jax/pull/2370 (by @hawkinsp) 

(See @mattjj 's comment https://github.com/google/jax/pull/2370#pullrequestreview-370624520: _"...The feedback from internal users was overwhelmingly in favor of this convention (for user code, and we should be consistent internally to avoid confusion)....The confusion can grow in a larger codebase that involves not only JAX code but also, say, TF code; the latter files may naturally have import numpy as np already. Another reason is that this is a convention followed by many other projects that present the NumPy API."_)

```diff
- import numpy as onp
+ import numpy as np
...
```

### CODE REVIEW

File: docs/annotated_mnist.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ import flax
 Load vanilla NumPy for use on host.
 
 ```py
-import numpy as onp
+import numpy as np
```
Comment (andsteing) [inline docs/annotated_mnist.md:11]: Do we actually use this anywhere in the example? Otherwise let's simply remove the entire paragraph.

Comment (8bitmp3) [reply docs/annotated_mnist.md:11]: Good point @andsteing

If NumPy isn't used, I'll make sure I remove the import

**Comment by avital** [comment]
This looks great @8bitmp3, thanks! Is it ready to be merged?

**Comment by 8bitmp3** [comment]
Done @avital. By the way, `distributed-training.diff` and `ensembling.diff` in `howtos/diff` are the only two files I haven't touched - not sure what to do. LMKWYT 👍

**Review by avital — APPROVED** [review]

**Comment by levskaya** [comment]
@8bitmp3 - hey sorry for this not going through yet.  There's a weird bug in our source sync system that makes the import script go nuts when there's over a certain number of SHAs/commits in a PR, and this PR hit it.

Any chance you could rebase on master and submit a squashed version of this PR?  (We also prefer to keep only a few commits per PR for cleaner commit-history's sake.)

**Comment by 8bitmp3** [comment]
@levskaya @avital Ok thanks, on it

**Comment by 8bitmp3** [comment]
@levskaya @avital I squeezed most of the commits into one large one 😬 Hope this doesn't break anything. I can't figure out what to fix in my merge conflicts though, please advise 👍

**Comment by levskaya** [comment]
This is superseded by a clean-merged version of the PR at #1027

**Comment by levskaya** [comment]
@8bitmp3 - would you mind adding your CLA consent to the updated (merged) version of this PR at #1027 ?

**Pull request closed without merging**